### PR TITLE
Antag tokens and Metacoins from Beestation

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Adds the metacoin system, which can be renamed to whatever one would wish in the config. Also adds the antag token system, which is used for tracking, well, how many times the person can exchange an antag token  for an antag round. Subject to the headmins' rulings, of course.

port of:
BeeStation/BeeStation-Hornet#85
BeeStation/BeeStation-Hornet#232
BeeStation/BeeStation-Hornet#363 (really only this, but the other two are relevant)

## Why It's Good For The Game
Creates incentive for playing for a while, as you are awarded metacoins for being on the server for a certain amount of time. Also, loadout system with paid items coming soon.

## Changelog
:cl:
add: Metacoin system. Get paid for greentexting! (or, you know, just surviving.)
add: antag tokens. Possibly get refunds for admins and griefing(?)
/:cl:

### CODE REVIEW

**Comment by triplezeta** [comment]
nigga bot nigga bot

**Comment by MarkSuckerberg** [comment]
github actions more like git the fuck out actions gottem

**Comment by SynnGraffkin** [comment]
boomer bucks when

**Comment by Crossedfall** [comment]
It'd be cool beans if you properly cherry-picked commits so that the original authorship remained, also a link back to the PR would be nice. Noticed you haven't been doing that at all. Very uncool. It's important to give credit to the people who put in the effort to create those features, both from our own codebase and other codebases you've ported from.

**Comment by MarkSuckerberg** [comment]
Ach, that's on me. I've been making barebones PR descriptions without doing much besides mentioning it's a port. As for the cherry-picking, I have been doing that, I'm just likely doing it wrong as the author information doesn't show up when I eventually do commit the changes. I'll go back and add the relevant PRs to all the content I have been porting, and I truly do apologize for any content that seems stolen. I didn't mean it to be that way.

**Comment by 1glitchycent** [comment]
@Crossedfall we already have the codebase credits here, scroll down to the bottom https://github.com/WaspStation/WaspStation-1.0

**Comment by Crossedfall** [comment]
> 
> 
> @Crossedfall we already have the codebase credits here, scroll down to the bottom [WaspStation/WaspStation-1.0](https://github.com/WaspStation/WaspStation-1.0)

OK? That doesn't address my concern even remotely. We're not talking about codebase credits or acknowledgements. It's about individual authorship and preserving copyright. Mark properly linking to PRs is a start.

**Comment by SynnGraffkin** [comment]
Let me know when you get a patent for making a bit of code on an open source game on a niche service. Otherwise, I don't give a damn.

**Comment by dinnerwoerror** [comment]
hey hold up, didn't ike try to pull this trick on some other guy

**Comment by triplezeta** [comment]
moffthink

**Comment by Crossedfall** [comment]
> 
> 
> Let me know when you get a patent for making a bit of code on an open source game on a niche service. Otherwise, I don't give a damn.

Annnd this is why the codebase isn't being taken seriously. 

> 
> 
> hey hold up, didn't ike try to pull this trick on some other guy

I have no idea.

The fact that there are people here seriously defending the idea of not crediting people for their hard work is honestly really disheartening.

**Comment by AsciiSquid** [comment]
The moral grandstanding isn't necessary. Mark has addressed your comment and has been working on correcting the error.

**Comment by Crossedfall** [comment]
> 
> 
> The moral grandstanding isn't necessary. Mark has addressed your comment and has been working on correcting the error.

Oh please. "Moral grandstanding?" Really? 

For Mark, he's fine in my view. He did the responsible and respectful thing and I applaud him on that. It's people coming into the thread undoing his good will that I take issue with now.

**Comment by SynnGraffkin** [comment]
@crossedfall again, you're talking about copywrites and code ownership in an open source codebase with code from other open source codebases. I would just say that having whatever you worked on ported to another server is good enough because another server actually liked what was made. Do you see me throwing a hissy fit at Citadel for using a PR I made for Hyper porting from AUstation? Fuck no. This shit ain't like Goon or other closed sources that need to screee whenever a good feature is ported. 

Maybe bully a codebase that will actually cave into your bullshit.

**Comment by Crossedfall** [comment]
> 
> 
> @Crossedfall again, you're talking about copywrites and code ownership in an open source codebase with code from other open source codebases. I would just say that having whatever you worked on ported to another server is good enough because another server actually liked what was made. Do you see me throwing a hissy fit at Citadel for using a PR I made for Hyper porting from AUstation? Fuck no. This shit ain't like Goon or other closed sources that need to screee whenever a good feature is ported.
> 
> Maybe bully a codebase that will actually cave into your bullshit.

I'm hardly bullying anyone. Mark has put in a lot of hard work and he has been wanting it to be recognized. If you all want to degrade whatever good will gained, then so be it. This isn't my project and I wont tell you all how to run it. But acting unprofessional and disrespectful is a good way to lose respect in the entire community.

**Comment by SynnGraffkin** [comment]
If being a dick works for the citrus man of TG, I don't think anyone will actually care what I say in comparison to him.

**Comment by AsciiSquid** [comment]
> It's people coming into the thread undoing his good will that I take issue with now.

That's fine, but turning around and acting like you can represent the voice and outlook of the entire community over a few comments you disagree with is incredibly arrogant and equally disrespectful. Discourse is fine, yet your attitude is unwarranted.

**Comment by Crossedfall** [comment]
> 
> 
> If being a dick works for the citrus man of TG, I don't think anyone will actually care what I say in comparison to him.

I said that was it for me, but I can't resist providing one final comment on this. The difference is size. /tg/ is large enough to where it doesn't matter as much. You don't have that luxury. Everyone involved has a massive influence on how people perceive you and the project as a whole. Maybe you should think on that. Credit where credit is due, based on your prior PRs you've at least given the bare minimum credits to the original authors by linking to the original PRs. So respect for that, but undoing that good will by making an ass of yourself wont take you very far. 

>
>
> That's fine, but turning around and acting like you can represent the voice and outlook of the entire community over a few comments you disagree with is incredibly arrogant and equally disrespectful. Discourse is fine, yet your attitude is unwarranted.

I think you're over inflating things a bit too much. I'm hardly acting like I'm some kind of mega representative of the community. My initial comment was from the perspective of a contributor. Anything after that has been my own views and opinions. Trying to turn it into something more than that is in itself arrogant. Stop trying to make this into something bigger than it is.

That really will do it for me though, since this has become massive off topic for the PR.

**Comment by SynnGraffkin** [comment]
Oh, I know what I say is not reflective of Mark and other contributors, but that's because I don't take bullshit lightly. You got a rando coming in here complaining and trying to bully my boy, and I ain't toleratin that. Only we can bully Marg, with love.

**Comment by Crossedfall** [comment]
> 
> 
> Oh, I know what I say is not reflective of Mark and other contributors, but that's because I don't take bullshit lightly. You got a rando coming in here complaining and trying to bully my boy, and I ain't toleratin that. Only we can bully Marg, with love.

Rando to you, maybe.

**Comment by SynnGraffkin** [comment]
Crossed: "Aight, I'mma head out. 
[I make a post] 
Crossed: "One more thing..."

**Comment by ike709** [comment]
> hey hold up, didn't ike try to pull this trick on some other guy

I did, in fact, ask Mark to put in the bare minimum effort when crediting his ports. Specifically in response to his previous series of ports from Oracle.

**Comment by dee-dubya** [comment]
I'm all for crediting people for their work, but here's a thought if you don't want people to use your code: Don't write a commit to an open source project.

**Comment by dinnerwoerror** [comment]
jesus christ y'all need to shut the hell up and expunge your penises from this absolute heap of a hamburger if you aren't gonna be productive

tl:dr i politely ask you to fuck off

**Comment by CthulhuOnIce** [comment]
The AGPL is a copyleft license that requires code to be credited to authors.

**Comment by SynnGraffkin** [comment]
@CthulhuOnIce If you wanna find the exact thing in the licensing for this, feel free to sort through all that legalese bullshit.

**Comment by Baldir-Odensen** [comment]
Now we just need qwerty and we got all the bee high ups

**Comment by dinnerwoerror** [comment]
alright, speaking from the position of an official Mentor of the BoomerStation13 administration, it's apparent that i wasn't clear enough the first time i told yous to fuck off, so let me explain it a second time

the man making this pull request has made his decision regarding the current topic of strife, and it appears that you wanna try and power play him to change it somehow

however, i think you people forgot that this is not beep station or Trans Gender station, you people hold no power here

the admins on boomer station have told this pull request creator what to do, not the admins on wherever you people fucking came from

if you want to hold power here than by all means, make the efforts to get it, otherwise, you people ought to go back to telling people what to do on YOUR own server, and us people will go back to telling people what to do on OUR own server

**Comment by dinnerwoerror** [comment]
tl;dr the pull request creator has chosen to acknowledge the misstep in erratum, yet you niggas are still here trying to "seize the moral high ground" or some shit for negligible gain

**Comment by optimumtact** [comment]
nice soapboxing.

**Comment by SynnGraffkin** [comment]
When Orange drops in to see the salt between two downstreams 

image

**Comment by steamp0rt** [comment]
@dinnerwoerror lmao you're on the side that's complaining because they got asked to give credit for code.

**Comment by MarkSuckerberg** [comment]
No offense to anyone here, but like, can we move on with our lives already? I've accepted responsibility for my error(s) and am working to remedy them. Nothing else really needs to be said on the topic, in my opinion.

**Comment by steamp0rt** [comment]
@MarkSuckerberg yeah it's kinda weird. You were like "oh okay, will fix" meanwhile like 3 nitwits show up to screech about... i don't know what, actually. they got offended by something that wasn't even targeted towards them

**Comment by MarkSuckerberg** [comment]
Comments not related to specifically this PR have been hidden. If you want to dispute the resolution of this, licensing terms, copypastas, whatever you wish, take it elsewhere. This is no place for that. I'm sorry to say, but if this continues, I will be locking the thread.

**Comment by gardenBanter** [comment]
Pee pee poo poo

**Comment by qwertyquerty** [comment]
I was here
